stages/org.osbuild.zip

```diff
@@ -0,0 +1,91 @@
+#!/usr/bin/python3
+"""
+Compresses an input file using zip.
+"""
+
+import os
+import subprocess
+import sys
+
+import osbuild.api
+
+CAPABILITIES = ["CAP_MAC_ADMIN"]
+
+
+SCHEMA_2 = """
+"options": {
+  "additionalProperties": false,
+  "required": ["filename"],
+  "properties": {
+    "filename": {
+      "description": "Filename for tar archive",
+      "type": "string"
+    },
+    "paths": {
+      "type": "array",
+      "items": {
+        "type": "string",
+        "description": "If specified, the archive will only contain the specified paths, in the specified order"
+      }
+    },
+    "level": {
+      "description": "Compression level",
+      "type": "integer",
+      "minimum": 0,
+      "maximum": 9,
+      "default": 6
+    }
+  }
+},
+"inputs": {
+  "type": "object",
+  "additionalProperties": false,
+  "required": ["tree"],
+  "properties": {
+    "tree": {
+      "type": "object",
+      "additionalProperties": true
+    }
+  }
+}
+"""
+
+
+def main(inputs, output_dir, options):
+    tree = inputs["tree"]["path"]
+    filename = options["filename"].lstrip("/")
+    level = options.get("level", 6)
+    paths = options.get("paths", [])
+
+    extra_args = []
```

Comment:
We are not passing anything to extra_args, if we need it, it should also be an option in the schema.

Reply:
This is a good point. I left it in for now to see if we thought of any `extra_args` that we would need to implement for this stage, but I can remove this and just mention it in the upstream PR.